openshot-qt: 2.1.0 -> 2.2.0

AndersonTorres opened the pull request:
Also, libopenshot: 0.1.2 -> 0.1.3

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

tohl commented:
Thanks for the changes.

> * https://github.com/NixOS/nixpkgs/pull/22710.diff 

There are a lot of changes in the diff without changes in the actual nix
expression.  Would it be possible to keep the unchanged things actually
unchanged?

For example, is this change necessary?

-  libopenshot = callPackage ./libopenshot.nix {};
+  libopenshot = callPackage ./libopenshot.nix { };

Or this?

-    license = licenses.gpl3Plus;
+    license = with licenses; gpl3Plus;

-    platforms = platforms.linux;
+    platforms = with platforms; linux;

It is always good to keep the diff minimal for everybody to see the real
changes and not all that formatting noise.  It saves everybody time.

For example:

-  propagatedBuildInputs = [
-    qt55.qtbase
-    qt55.qtmultimedia
-    libopenshot
-  ];
+  propagatedBuildInputs =
+  [ libopenshot qt5.qtbase qt5.qtmultimedia ];

If you change qt55 to qt5, I should see no change to libopenshot.
However, in your diff, there is a change which I have to pay attention
to and there will be in the future, e.g. if somebody changes qt5 to qt.

Or this:

-  buildInputs = [
-    cmake doxygen
-    imagemagick ffmpeg qt55.qtbase qt55.qtmultimedia swig python3 ruby
-    unittest-cpp cppzmq czmqpp
+  buildInputs =
+  [ pkgconfig cmake doxygen imagemagick ffmpeg qt55.qtbase qt55.qtmultimedia
+    swig python3 ruby unittest-cpp cppzmq czmqpp

Did something change in regard with cmake, doxygen, unittest-cpp cppzmq
czmqpp?  If no, why does it need to be in the diff?

This doesn't bring anything either, or does it?

-  openshot-qt = callPackage ../applications/video/openshot-qt { };
+  openshot-qt = callPackage ../applications/video/openshot-qt { qt5 = qt55; };

tohl commented:
Also, you changed qt55 to qt5 but your build inputs still use qt55.

-  buildInputs = [
-    cmake doxygen
-    imagemagick ffmpeg qt55.qtbase qt55.qtmultimedia swig python3 ruby
-    unittest-cpp cppzmq czmqpp
+  buildInputs =
+  [ pkgconfig cmake doxygen imagemagick ffmpeg qt55.qtbase qt55.qtmultimedia
+    swig python3 ruby unittest-cpp cppzmq czmqpp

So, what changed here actually?  A new dependency, or removed
dependency, or just noise?

AndersonTorres commented:
Oh, my mistakes.
Yes, there are some cosmetic changes, that doesn't change the output at all, but make the code more uniform and easier to read. I don't know about documented strict guidelines of style for Nixpkgs expressions, but I invariably follow some templates: spaces before/after braces, merge lines, rearrange parameters etc.

And yes, I have forgotten to save some pieces of work, like changing all qt55 pieces. But I have tested with qt57 and openshot runs fine. I will fix this now.

grahamc commented:
The packages should be updated in separate commits, but I think it is ready to merge after that.

AndersonTorres commented:
When you said "separate commits", it is about updating every component? First libopenhot and after the openshot app itself?

peterhoeg commented:
The proper way to build the qt packages is to use ```qt5.callPackage``` instead (or qt55.callPackage if you need 5.5) which will give you direct access to ```qtbase``` and ```qtmultimedia```.

AndersonTorres commented:
I can see I'm a bit outdated on this things! I will refactor all now (or tonight!).

grahamc commented:
Can we get a re-review, @7c6f434c?

peterhoeg commented on pkgs/applications/video/openshot-qt/libopenshot-audio.nix:
```diff
@@ -14,18 +14,22 @@ stdenv.mkDerivation rec {
+  buildInputs =
+  [ pkgconfig cmake doxygen alsaLib
+    libX11 libXft libXrandr libXinerama libXext libXcursor ];
 
```
cmake and pkgconfig should go into ```nativeBuildInputs```.

peterhoeg commented on pkgs/applications/video/openshot-qt/default.nix:
```diff
@@ -1,29 +1,26 @@
 
-  buildInputs = [doxygen python3Packages.python makeWrapper ffmpeg];
+  buildInputs = [ doxygen python3Packages.python ffmpeg makeWrapper ];
 
```
cmake and doxygen (probably)  should go into nativeBuildInputs.

AndersonTorres replied:
OK, I will do it now! (Sorry by the delay...)

7c6f434c commented:
@grahamc you meant @tohl not me, right?

peterhoeg commented:
Couple of pointers:

 - you probably want ```wrapQtProgram``` instead of ```wrapProgram```
 - ```qt5.callPackage``` didn't work because the default version used to be 5.6 and openshot requires 5.7
 - default qt is now 5.7
 - ```qt.callPackage``` has changed to ```libsForQt5.callPackage``` (as well as libsForQt55, 56 and 57)

AndersonTorres commented:
No way. `libsForQt5.callPackage` is constantly failing, even the most recent version.

tohl commented:
I just removed myself from the maintainer list as I've settled on the
cli-based melt.  If anybody wants, please feel free to take over the
maintenance.

AndersonTorres commented:
This is the error message when I use qt5 wrappers:

> 
> + source /home/atorres/scripts/nix-scripts/common-vars.sh
> ++ NIXPKGS=/Attic/Programming/github/nixpkgs
> ++ LOCALINST=/home/atorres/fakeroot/nixpkgs
> ++ OPTIONS='--pure --show-trace'
> ++ OUTPUTS=/home/atorres/fakeroot/nixpkgs/outputs
> ++ DERIVATIONS=/home/atorres/fakeroot/nixpkgs/derivations
> + BUILDME=openshot-qt
> + DEFAULTS=' --pure --option build-use-sandbox true'
> + nix-build --run-env --pure --option build-use-sandbox true --pure --show-trace /Attic/Programming/github/nixpkgs -A openshot-qt --out-link /home/atorres/fakeroot/nixpkgs/outputs/openshot-qt --drv-link /home/atorres/fakeroot/nixpkgs/derivations/openshot-qt
> these derivations will be built:
>   /nix/store/lpqk5l7j7881zly6m6p528m30nbsrh3c-libopenshot-0.1.3.drv
> building path(s) ‘/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3’
> unpacking sources
> unpacking source archive /nix/store/3ildzb3jci78xp9z4m5r1cp21x34d1p9-libopenshot-v0.1.3-src
> source root is libopenshot-v0.1.3-src
> patching sources
> configuring
> fixing cmake files...
> cmake flags: -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3/include -DCMAKE_INSTALL_LIBDIR=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3/lib -DCMAKE_INSTALL_PREFIX=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3  
> --------------------------------------------------------------
> Welcome to the OpenShot Build System! CMake will now check for all required build
> dependencies and notify you of any missing files or other issues. If you have any
> questions or issues, please visit <http://www.openshot.org/>.
> --------------------------------------------------------------
> Determining Version Number (from Version.h file)
> --> MAJOR Version: 0
> --> MINOR Version: 1
> --> BUILD Version: 3
> --> SO/API/ABI Version: 0
> --> VERSION: 0.1.3
> 
> CMake Warning (dev) at CMakeLists.txt:66 (PROJECT):
>   Policy CMP0048 is not set: project() command manages VERSION variables.
>   Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
>   command to set the policy and suppress this warning.
> 
>   The following variable(s) would be set to empty:
> 
>     PROJECT_VERSION
> This warning is for project developers.  Use -Wno-dev to suppress it.
> 
> -- The C compiler identification is GNU 5.4.0
> -- The CXX compiler identification is GNU 5.4.0
> -- Check for working C compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/gcc
> -- Check for working C compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/gcc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/g++
> -- Check for working CXX compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/g++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> --------------------------------------------------------------
> Generating build files for openshot (0.1.3)
> -- Found ImageMagick: /nix/store/4hcja2f3dszjjdwxqyvjrk4cr3xcmv3m-imagemagick-6.9.7-6/lib/libMagick++-6.Q16.so (found version "6.9.7-6") 
> -- Found FFMPEG: /nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavformat.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavcodec.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavutil.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavdevice.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libswscale.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavresample.so  
> /nix/store/lfk2dz1l91b7fdznlnw6id1d2pcvfxm9-libopenshot-audio-0.1.2
> -- Found LIBOPENSHOT_AUDIO: /nix/store/lfk2dz1l91b7fdznlnw6id1d2pcvfxm9-libopenshot-audio-0.1.2/lib/libopenshot-audio.so  
> 'LIBOPENSHOT_AUDIO_INCLUDE_DIRS:/nix/store/lfk2dz1l91b7fdznlnw6id1d2pcvfxm9-libopenshot-audio-0.1.2/include/libopenshot-audio'
> CMake Error at src/CMakeLists.txt:98 (find_package):
>   By not providing "FindQt5Widgets.cmake" in CMAKE_MODULE_PATH this project
>   has asked CMake to find a package configuration file provided by
>   "Qt5Widgets", but CMake did not find one.
> 
>   Could not find a package configuration file provided by "Qt5Widgets" with
>   any of the following names:
> 
>     Qt5WidgetsConfig.cmake
>     qt5widgets-config.cmake
> 
>   Add the installation prefix of "Qt5Widgets" to CMAKE_PREFIX_PATH or set
>   "Qt5Widgets_DIR" to a directory containing one of the above files.  If
>   "Qt5Widgets" provides a separate development package or SDK, be sure it has
>   been installed.
> 
> 
> -- Configuring incomplete, errors occurred!
> See also "/tmp/nix-build-libopenshot-0.1.3.drv-0/libopenshot-v0.1.3-src/build/CMakeFiles/CMakeOutput.log".
> builder for ‘/nix/store/lpqk5l7j7881zly6m6p528m30nbsrh3c-libopenshot-0.1.3.drv’ failed with exit code 1
> error: build of ‘/nix/store/lpqk5l7j7881zly6m6p528m30nbsrh3c-libopenshot-0.1.3.drv’ failed
> /nix/var/nix/profiles/per-user/atorres/fullEnv/bin/nix-build: failed to build all dependencies
> {17-03-11 13:00}delorean-nixos:/tmp/test-nixpkgs atorres% OPTIONS='--pure --show-trace' sh ~/scripts/nix-scripts/run-nix-shell.sh openshot-qt
> + source /home/atorres/scripts/nix-scripts/common-vars.sh
> ++ NIXPKGS=/Attic/Programming/github/nixpkgs
> ++ LOCALINST=/home/atorres/fakeroot/nixpkgs
> ++ OPTIONS='--pure --show-trace'
> ++ OUTPUTS=/home/atorres/fakeroot/nixpkgs/outputs
> ++ DERIVATIONS=/home/atorres/fakeroot/nixpkgs/derivations
> + BUILDME=openshot-qt
> + DEFAULTS=' --pure --option build-use-sandbox true'
> + nix-build --run-env --pure --option build-use-sandbox true --pure --show-trace /Attic/Programming/github/nixpkgs -A openshot-qt --out-link /home/atorres/fakeroot/nixpkgs/outputs/openshot-qt --drv-link /home/atorres/fakeroot/nixpkgs/derivations/openshot-qt
> these derivations will be built:
>   /nix/store/lpqk5l7j7881zly6m6p528m30nbsrh3c-libopenshot-0.1.3.drv
> building path(s) ‘/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3’
> unpacking sources
> unpacking source archive /nix/store/3ildzb3jci78xp9z4m5r1cp21x34d1p9-libopenshot-v0.1.3-src
> source root is libopenshot-v0.1.3-src
> patching sources
> configuring
> fixing cmake files...
> cmake flags: -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3/include -DCMAKE_INSTALL_LIBDIR=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3/lib -DCMAKE_INSTALL_NAME_DIR=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3/lib -DCMAKE_INSTALL_PREFIX=/nix/store/0ssng3xlfrsqgdif5ap1v8rm99f83lgq-libopenshot-0.1.3  
> --------------------------------------------------------------
> Welcome to the OpenShot Build System! CMake will now check for all required build
> dependencies and notify you of any missing files or other issues. If you have any
> questions or issues, please visit <http://www.openshot.org/>.
> --------------------------------------------------------------
> Determining Version Number (from Version.h file)
> --> MAJOR Version: 0
> --> MINOR Version: 1
> --> BUILD Version: 3
> --> SO/API/ABI Version: 0
> --> VERSION: 0.1.3
> 
> CMake Warning (dev) at CMakeLists.txt:66 (PROJECT):
>   Policy CMP0048 is not set: project() command manages VERSION variables.
>   Run "cmake --help-policy CMP0048" for policy details.  Use the cmake_policy
>   command to set the policy and suppress this warning.
> 
>   The following variable(s) would be set to empty:
> 
>     PROJECT_VERSION
> This warning is for project developers.  Use -Wno-dev to suppress it.
> 
> -- The C compiler identification is GNU 5.4.0
> -- The CXX compiler identification is GNU 5.4.0
> -- Check for working C compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/gcc
> -- Check for working C compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/gcc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/g++
> -- Check for working CXX compiler: /nix/store/wgvrkijby0awhqkyfkj0qp9cc4921hd9-gcc-wrapper-5.4.0/bin/g++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> --------------------------------------------------------------
> Generating build files for openshot (0.1.3)
> -- Found ImageMagick: /nix/store/4hcja2f3dszjjdwxqyvjrk4cr3xcmv3m-imagemagick-6.9.7-6/lib/libMagick++-6.Q16.so (found version "6.9.7-6") 
> -- Found FFMPEG: /nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavformat.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavcodec.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavutil.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavdevice.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libswscale.so;/nix/store/xyqg2hync0mgkbr730v3y9z4279qx6v5-ffmpeg-3.1.7/lib/libavresample.so  
> /nix/store/lfk2dz1l91b7fdznlnw6id1d2pcvfxm9-libopenshot-audio-0.1.2
> -- Found LIBOPENSHOT_AUDIO: /nix/store/lfk2dz1l91b7fdznlnw6id1d2pcvfxm9-libopenshot-audio-0.1.2/lib/libopenshot-audio.so  
> 'LIBOPENSHOT_AUDIO_INCLUDE_DIRS:/nix/store/lfk2dz1l91b7fdznlnw6id1d2pcvfxm9-libopenshot-audio-0.1.2/include/libopenshot-audio'
> CMake Error at src/CMakeLists.txt:98 (find_package):
>   By not providing "FindQt5Widgets.cmake" in CMAKE_MODULE_PATH this project
>   has asked CMake to find a package configuration file provided by
>   "Qt5Widgets", but CMake did not find one.
> 
>   Could not find a package configuration file provided by "Qt5Widgets" with
>   any of the following names:
> 
>     Qt5WidgetsConfig.cmake
>     qt5widgets-config.cmake
> 
>   Add the installation prefix of "Qt5Widgets" to CMAKE_PREFIX_PATH or set
>   "Qt5Widgets_DIR" to a directory containing one of the above files.  If
>   "Qt5Widgets" provides a separate development package or SDK, be sure it has
>   been installed.
> 
> 
> -- Configuring incomplete, errors occurred!
> See also "/tmp/nix-build-libopenshot-0.1.3.drv-0/libopenshot-v0.1.3-src/build/CMakeFiles/CMakeOutput.log".
> builder for ‘/nix/store/lpqk5l7j7881zly6m6p528m30nbsrh3c-libopenshot-0.1.3.drv’ failed with exit code 1
> error: build of ‘/nix/store/lpqk5l7j7881zly6m6p528m30nbsrh3c-libopenshot-0.1.3.drv’ failed
> /nix/var/nix/profiles/per-user/atorres/fullEnv/bin/nix-build: failed to build all dependencies
>

peterhoeg commented:
You need to pass in ```qtbase``` to ```buildInputs```.

AndersonTorres commented:
I think it is now over. More thoughts?

AndersonTorres committed:
openshot-qt: 2.1.0 -> 2.2.0

Also, libopenshot and libopenshot-audio
are now directly referenced on all-packages.nix.

AndersonTorres commented:
Now it is compiling fine. Can we close it and merge?
PING @peterhoeg @grahamc

peterhoeg commented on pkgs/applications/video/openshot-qt/default.nix:
```diff
@@ -1,26 +1,28 @@
+  
+  buildInputs =
+  [ python3Packages.python ffmpeg qtbase qtmultimedia makeQtWrapper ];
 
```
makeQtWrapper should be in ```nativeBuildInputs```.

peterhoeg commented on pkgs/applications/video/openshot-qt/default.nix:
```diff
@@ -1,26 +1,28 @@
-    libopenshot
-  ];
+  propagatedBuildInputs =
+  [ libopenshot ];
```
Does it work with libopenshot in ```buildInputs```?

AndersonTorres replied:
Yes, I have changed to reflect this.

AndersonTorres commented:
Many thanks, @peterhoeg !

peterhoeg commented:
No worries, thanks for your patience. Good we got this merged in time for the release of openshot 2.3 y'day! ;-)
